packages/algob/src/lib/asc.ts

```diff
@@ -0,0 +1,33 @@
+import algosdk from "algosdk";
+
+import {
+  ASCDeploymentFlags
+} from "../types";
+
+async function getSuggestedParams (algocl: algosdk.Algodv2): Promise<algosdk.SuggestedParams> {
+  const params = await algocl.getTransactionParams().do();
+  // Private chains may have an issue with firstRound
+  if (params.firstRound === 0) {
+    throw new Error("Suggested params returned 0 as firstRound. Ensure that your node progresses.");
+    // params.firstRound = 1
+  }
+  return params;
+}
+
+export async function getSuggestedParamsWithUserDefaults (
+  algocl: algosdk.Algodv2, userDefaults: ASCDeploymentFlags): Promise<algosdk.SuggestedParams> {
+  const suggested = await getSuggestedParams(algocl);
```

Comment:
Are these params from the spec? You simply copied my code for ASA.

Reply:
yes, These parameters are for payment transaction which is same for all,
- for funding smart contract from creator account.